### PR TITLE
1266 check search criteria

### PR DESCRIPTION
**Pull request by btry**
Prevent HTTP 500 when using a non existent criteria field ID

Also fix a comment

### CODE REVIEW

**Comment by btry** [comment]
I found tests on the API where HTTP response shoud be 4xx are buggy. try / catch block never catch exception because it is not re-trhown by APIRestTest::doHttpRequest()

I needed to fix this to implement my own test

File: tests/API/APIRestTest.php [hunk 1]
```diff
@@ -423,18 +441,20 @@ public function testBadEndpoint($session_token) {
          $res = $this->doHttpRequest('GET', 'badEndpoint/',
                                             ['headers' => [
                                              'Session-Token' => $session_token]]);
+         $this->assertFalse(true); // if HTTP 2xx then the test must fail
```
Comment (orthagh) [inline tests/API/APIRestTest.php]: i think i would prefer 

$this->assertLessThan(300, $response->getStatusCode());

File: tests/API/APIRestTest.php [hunk 2]
```diff
@@ -423,18 +441,20 @@ public function testBadEndpoint($session_token) {
...
          $response = $e->getResponse();
-         $this->assertEquals(400, $response->getStatusCode());
+         $this->assertEquals(400, $this->last_error->getStatusCode());
       }
 
       try {
          $res = $this->doHttpRequest('GET', 'Entity/0/badEndpoint/',
                                             ['headers' => [
                                              'Session-Token' => $session_token]]);
+         $this->assertFalse(true); // if HTTP 2xx then the test must fail
```
Comment (orthagh) [inline tests/API/APIRestTest.php]: same

File: tests/API/APIRestTest.php [hunk 3]
```diff
@@ -761,9 +781,10 @@ public function testgetItemsForPostonly() {
          $res = $this->doHttpRequest('GET', "ticket/$tickets_id",
                                             ['headers' => [
                                                 'Session-Token' => $data['session_token']]]);
+         $this->assertFalse(true); // if HTTP 2xx then the test must fail
```
Comment (orthagh) [inline tests/API/APIRestTest.php]: same

File: inc/api.class.php [hunk 4]
```diff
@@ -1295,6 +1295,15 @@ protected function searchItems($itemtype, $params=array()) {
       // retrieve searchoptions
       $soptions = $this->listSearchOptions($itemtype);
 
+      // Check the criterias are valid
+      if (isset($params['criteria']) && is_array($params['criteria'])) {
+         foreach ($params['criteria'] as $criteria) {
+            if (isset($criteria['field']) && ctype_digit($criteria['field']) && !array_key_exists($criteria['field'], $soptions)) {
+               return $this->returnError("Bad field ID in search criteria");
```
Comment (btry) [inline inc/api.class.php]: oops ! sorry

File: tests/API/APIRestTest.php (hunk 1 above)
Comment (btry) [inline tests/API/APIRestTest.php]: I understand your goal, but if the response is 200 the test will succeed and this is not what we want.

Maybe  the following would be more appropriate ?
$this->assertGreaterThanOrEqual(400, $res->getStatusCode());

**Pull request merged**